FlyingSocks/Sources/Socket.swift

```diff
@@ -237,6 +266,29 @@ public struct Socket: Sendable, Hashable {
         return sent
     }
 
+    public func send(_ bytes: [UInt8], to address: some SocketAddress) throws -> Int {
+        try bytes.withUnsafeBytes { buffer in
+            try send(buffer.baseAddress!, length: bytes.count, to: address)
+        }
+    }
+
+    private func send(_ pointer: UnsafeRawPointer, length: Int, to address: some SocketAddress) throws -> Int {
+        var addr = address
+        let sent = withUnsafePointer(to: &addr) {
+            $0.withMemoryRebound(to: sockaddr.self, capacity: 1) {
+                Socket.sendto(file.rawValue, pointer, length, 0, $0, address.size)
+            }
+        }
+        guard sent >= 0 || errno == EISCONN else {
+            if errno == EINPROGRESS {
```

Comment:
Is `EINPROGRESS` the correct error here or should it be `EWOULDBLOCK` ?

With `SOCK_STREAM` you can test this by setting a tiny write buffer and writing more data than is read from the other end:
```swift
try socket.setValue(5, for: .sendBufferSize)
try socket.write(Data(repeating: 0x01, count: 100))
```

But with `SOCK_DGRAM` I can't get it to ever block, I suppose it just drops?